Content.Server/Body/Systems/RespiratorSystem.cs

```diff
@@ -184,44 +193,64 @@ public bool CanMetabolizeInhaledAir(Entity<RespiratorComponent?> ent)
         if (!Resolve(ent, ref ent.Comp))
             return false;
 
-        if (!Inhale(ent))
+        // Get the gas at our location but don't actually remove it from the gas mixture.
+        var ev = new InhaleLocationEvent
+        {
+            Respirator = ent.Comp,
+        };
+        RaiseLocalEvent(ent, ref ev);
+
+        ev.Gas ??= _atmosSys.GetContainingMixture(ent.Owner, excite: true);
+
+        // If there's no air to breathe or we can't metabolize it then internals should be on.
+        return ev.Gas is not null && CanMetabolizeInhaledAir(ent, ev.Gas);
+    }
+
+    public bool CanMetabolizeInhaledAir(Entity<RespiratorComponent?> ent, GasMixture gas)
+    {
+        if (!Resolve(ent, ref ent.Comp))
+            return false;
+
+        // Properly try to inhale the given gas mixture
+        var inhaleEv = new InhaledGasEvent(gas);
+        RaiseLocalEvent(ent, ref inhaleEv);
+
+        if (!inhaleEv.Handled || !inhaleEv.Succeeded)
             return false;
 
-        // If we don't have a body we can't be poisoned by gas, yet...
-        var success = TryMetabolizeGas((ent, ent.Comp));
+        // Exhaling will give the gas back so we need to remove the gas from the mixture.
+        // If it doesn't give the gas back content integration tests should fail.
+        gas.RemoveVolume(ent.Comp.BreathVolume);
+
+        var success = CanMetabolizeGas((ent, ent.Comp));
+
+        Exhale(ent, gas);
 
-        // Don't keep that gas in our lungs lest it poisons a poor nuclear operative.
-        Exhale(ent);
         return success;
     }
 
     /// <summary>
-    /// Check whether or not an entity can metabolize the given gas mixture without suffocating or taking damage
-    /// (i.e., no toxic gasses).
+    /// Tries to safely metabolize the current solutions in a body's lungs.
     /// </summary>
-    public bool CanMetabolizeGas(Entity<RespiratorComponent?> ent, GasMixture gas)
+    private bool CanMetabolizeGas(Entity<RespiratorComponent, BodyComponent?> ent)
     {
-        if (!Resolve(ent, ref ent.Comp))
+        if (!Resolve(ent, ref ent.Comp2))
```

Comment:
isn't this just coupling body and respirator again? This will error if it fails to resolve and it's used in the normal control flow as a backup option